core/api/src/classes/loggedModel.ts

```diff
@@ -146,4 +146,24 @@ export abstract class LoggedModel<T> extends Model<T> {
   }
 
   abstract apiData(): Promise<{ [key: string]: any }>;
+
+  /**
+   * Find an instance of this class, regardless of scope
+   */
+  static async findByGuid(guid: string): Promise<any> {
+    // static class definitions or type defining are not yet available in TS.  See:
+    // * https://github.com/microsoft/TypeScript/issues/14600
+    // * https://github.com/microsoft/TypeScript/issues/34516
+    // * https://github.com/microsoft/TypeScript/issues/33892
+
+    // So, each model will implement this method
+
+    throw new Error("not implemented");
+
+    // const instance: T = await this.scope(null).findOne({ where: { guid } });
+    // if (!instance) {
+    //   throw new Error(`cannot find ${this.name} ${guid}`);
+    // }
+    // return instance;
+  }
```

Comment:
If we can define this here we won't need to on every model... I don't think it's possible though at the moment in TS.

Reply:
what! boo.
I assume the issue is knowing that it's `Group.scope...` vs. `Destination.scope...`
What if you made this method do
```
const klass = this.klass // or this.klas();
const instance: T = await klass.scope(null).findOne({ where: { guid } });
if (!instance) {
  throw new Error(`cannot find ${klass.name} ${guid}`);
}
return instance;
```
And then implemented `this.klass = ` or `function klass()` in the children. I bet that would pay off on the next thing somewhat like this.

Reply:
Related: when I make the Group class it does this: `export class Group extends LoggedModel<Group>`
Doesn't the code have access to that `<Group>` thing somewhere?

Reply:
You would think so...
<img width="642" alt="Screen Shot 2020-04-15 at 3 46 51 PM" src="https://user-images.githubusercontent.com/303226/79396250-546feb00-7f30-11ea-85d5-fa3b09603d38.png">